src/main/java/relay/data_access/FirebaseCourseDataAccessObject.java

```diff
@@ -0,0 +1,114 @@
+
+package relay.data_access;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.*;
+
+import relay.entity.Course;
+import relay.entity.Instructor;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+
+public class FirebaseCourseDataAccessObject {
+
+
+    /**
+     * Creates a new course document in Firestore based on the given Course object.
+     *
+     * @param course The Course object containing course details to be added to Firestore.
+     */
+    public void createCourse(Course course) {
+        Map<String, Object> courseDocument = new HashMap<>();
+        courseDocument.put("courseID", course.getCourseID());
```

Comment:
This shouldn't be taken from the provided course since a new course will not have a Firebase auto-generated courseID. Leave the courseID blank. Firebase course documents should also not have a courseID parameter.